nservicebus/messaging/message-type-detection.md

```diff
@@ -1,24 +1,24 @@
 ---
 title: Message type detection
 summary: Describes the rules for mapping a transport message to a message type
-reviewed: 2021-06-05
+reviewed: 2023-08-04
 component: Core
+related:
+- samples/consumer-driven-contracts
 ---
 
-In order to invoke the correct message handlers for incoming messages, NServiceBus must be able to map the incoming transport message to a [message type](/nservicebus/messaging/messages-events-commands.md).
+NServiceBus sets the [`NServiceBus.EnclosedMessageTypes` header](/nservicebus/messaging/headers.md#serialization-headers-nservicebus-enclosedmessagetypes) to the [AssemblyQualifiedName](https://docs.microsoft.com/en-us/dotnet/api/system.type.assemblyqualifiedname) of the emitted message type when sending, publishing or replying.
+
+It is possible to change the enclosed message type header for all emitted messages by using a behavior. The [consumder-driven-contracts sample](/samples/consumer-driven-contracts) shows how to manipulate the header for outgoing messages.
+
+To invoke the correct message handlers for incoming messages, NServiceBus must be able to map the incoming transport message to a [message type](/nservicebus/messaging/messages-events-commands.md).
 
 The mapping rules are as follows:
 
 1. If the message contains the [`NServiceBus.EnclosedMessageTypes` header](/nservicebus/messaging/headers.md#serialization-headers-nservicebus-enclosedmessagetypes), the header value is used to find the message type. The header value may contain:
+   - The [AssemblyQualifiedName](https://docs.microsoft.com/en-us/dotnet/api/system.type.assemblyqualifiedname) of the message type (with or without private key are both supported). In cases when the assembly qualified type is not known by the endpoint NServiceBus will fall back using the [FullName](https://docs.microsoft.com/en-us/dotnet/api/system.type.fullname) of the message type.
```

Comment:
Feels redundant given the fullname is the next point in the list?

```suggestion
   - The [AssemblyQualifiedName](https://docs.microsoft.com/en-us/dotnet/api/system.type.assemblyqualifiedname) of the message type (with or without private key are both supported).
```

Reply:
No. This addition is crucial to address the feedback of the linked issue. I will reword it a bit